### PR TITLE
Moving PersistentLocalVolumes feature gate check from validation

### PR DESCRIPTION
**Pull request by sbezverk**
Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>
xref: #72651 

/release-note-none
/sig api-machinery
/sig storage
/kind bug

```release-note
none
```

### CODE REVIEW

File: pkg/api/persistentvolume/util_test.go [hunk 1]
```diff
@@ -27,7 +28,7 @@ import (
 	"k8s.io/kubernetes/pkg/features"
 )
 
-func TestDropDisabledFields(t *testing.T) {
+func TestDropDisabledFieldsCSI(t *testing.T) {
```
Comment (liggitt) [inline pkg/api/persistentvolume/util_test.go]: this was also testing block mode... revert the name change?

File: pkg/api/persistentvolume/util_test.go [hunk 2]
```diff
@@ -152,3 +153,99 @@ func TestDropDisabledFields(t *testing.T) {
 		})
 	}
 }
+
+func TestDropDisabledFieldsPersistentLocalVolume(t *testing.T) {
+	fsType := "ext4"
```
Comment (liggitt) [inline pkg/api/persistentvolume/util_test.go]: move inside pvWithLocalVolume

**Comment by liggitt** [comment]
two nits, lgtm otherwise

File: pkg/api/persistentvolume/util_test.go (hunk 1 above)
Comment (sbezverk) [reply pkg/api/persistentvolume/util_test.go]: done

File: pkg/api/persistentvolume/util_test.go (hunk 2 above)
Comment (sbezverk) [reply pkg/api/persistentvolume/util_test.go]: done

**Comment by liggitt** [comment]
@kubernetes/sig-storage-api-reviews 

/lgtm
/approve

**Comment by liggitt** [comment]
\o/

**Pull request merged**